Source/Dafny/Dafny.atg

```diff
@@ -2037,21 +2039,31 @@ TypeAndToken<out IToken tok, out Type ty, bool inExpressionContext>
 
 /*------------------------------------------------------------------------*/
 
-TupleType<.out Type ty, out IToken tok, out List<Type> tupleArgTypes.> =
+TupleType<.out Type ty, out IToken tok, out List<Type> tupleArgTypes, out List<bool> ghostArguments.> =
   "("                             (. tok = t;
                                      ty = null; // To keep compiler happy
                                      tupleArgTypes = new List<Type>();
+                                     var isGhost = false;
+                                     ghostArguments = new List<bool>();
                                   .)
-  [ Type<out ty>                  (. tupleArgTypes.Add(ty); .)
-    { "," Type<out ty>            (. tupleArgTypes.Add(ty); .)
+  [ [ "ghost"                     (. isGhost = true; .)
+    ]
+    Type<out ty>                  (. tupleArgTypes.Add(ty); ghostArguments.Add(isGhost); .)
+    { ","                         (. isGhost = false; .)
+      [ "ghost"                   (. isGhost = true; .)
+      ]
+      Type<out ty>                (. tupleArgTypes.Add(ty); ghostArguments.Add(isGhost); .)
```

Comment:
There needs to be a corresponding change to the `_openparen` case of method `IsNonArrowType` in `Dafny.atg`. See the comment in method `IsGenericInstantiation` in `Dafny.atg`, and also see module `M1` in `Test/dafny0/UserSpecifiedTypeParameters.dfy`.

Reply:
I see, the `IsTypeSequence` method needs to match `[ "ghost" ] Type { "," [ "ghost" ] Type }` instead of `Type { "," Type }`.